### PR TITLE
applications: nrf5340_audio: Fix for receiving capabilities

### PR DESCRIPTION
**Pull request by erikrobstad**
Storing the capabilities correctly

### CODE REVIEW

**Review by alexsven — APPROVED** [review]

**Review by rick1082 — APPROVED** [review]

File: applications/nrf5340_audio/src/bluetooth/le_audio_cis_gateway.c [hunk 1]
```diff
@@ -76,6 +76,7 @@ struct temp_cap_storage {
 	struct bt_conn *conn;
 	/* Must be the same size as sink_codec_cap and source_codec_cap */
 	struct bt_codec codec[CONFIG_CODEC_CAP_COUNT_MAX];
+	uint8_t num_caps;
```
Comment (koffes) [inline applications/nrf5340_audio/src/bluetooth/le_audio_cis_gateway.c:79]: Ref. the comment on line 802-804: We can make this clearer by 
```suggestion
	uint8_t num_caps_total;
```

File: applications/nrf5340_audio/src/bluetooth/le_audio_cis_gateway.c [hunk 2]
```diff
@@ -797,6 +798,11 @@ static void discover_sink_cb(struct bt_conn *conn, struct bt_codec *codec, struc
 			for (int i = 0; i < codec->meta_count; i++) {
 				store_loc->meta[i].data.data = store_loc->meta[i].value;
 			}
+
+			/* In temp_cap.num_caps we want to store the actual number
+			 * of capabilities, while params->num_caps is an iterator
+			 */
+			temp_cap[temp_cap_index].num_caps = params->num_caps + 1;
```
Comment (koffes) [inline applications/nrf5340_audio/src/bluetooth/le_audio_cis_gateway.c:805]: Should there be a check if num_caps exceeds CONFIG_CODEC_CAP_COUNT_MAX?

Comment (koffes) [reply applications/nrf5340_audio/src/bluetooth/le_audio_cis_gateway.c:805]: Here and on line 933

File: applications/nrf5340_audio/src/bluetooth/le_audio_cis_gateway.c (hunk 1 above)
Comment (erikrobstad) [reply applications/nrf5340_audio/src/bluetooth/le_audio_cis_gateway.c:79]: Good point, done

File: applications/nrf5340_audio/src/bluetooth/le_audio_cis_gateway.c (hunk 2 above)
Comment (erikrobstad) [reply applications/nrf5340_audio/src/bluetooth/le_audio_cis_gateway.c:805]: params->num_caps is inderectly cheched against CONFIG_CODEC_CAP_COUNT_MAX on line 788. temp_cap[temp_cap_index].codec has the length of CONFIG_CODEC_CAP_COUNT_MAX

**Review by koffes — APPROVED** [review]

**Pull request merged**